solutions/usaco-671.mdx

```diff
@@ -84,4 +84,64 @@ int main()
 
 </CPPSection>
 
+<PySection>
+
+```py
+import sys
+from collections import defaultdict, deque
+sys.stdin, sys.stdout = open('lasers.in', 'r'), open('lasers.out', 'w')
```

Comment:
assign these on separate lines

Reply:
actually i'd rather you use `with` blocks instead of redirecting `stdin` & `stdout`

Reply:
Would something like this be fine?
```py
def main(infile, outfile):
	... # solve here

with open('lasers.in') as infile, open('lasers.out') as outfile:
	main(infile, outfile)
```

Reply:
use 2 separate with blocks

Reply:
not sure if I did it correctly but I did what seemed most reasonable to me